explainer-strings.md

```diff
@@ -0,0 +1,163 @@
+very nice writeup by the security researcher who discovered it](https://research.securitum.com/mutation-xss-via-mathml-mutation-dompurify-2-0-17-bypass/).
+For details, we encourage the reader to follow the link
+
+The gist is that this string will change its meaning when being re-parsed. It
+has been carefully crafted to have one interpretation when parsed, and another
+one when being re-parsed, after having been parsed and un-parsed previously.
+Note that it contains a `<style>` element. On the initial parse, the `<img>`
+element, which contains an event handler, is inside the `<style>` element and
+is thus being parsed as a (malformed) stylesheet. On the second parse, it
+will be parsed as a regular element, with an event handler.
+
+If this example is being presented to a sanitizer library, the sanitizer will
+see the result of the first pass, and will let the "event handler" - which it
+sees as style-sheet content - pass. If this is being re-parsed on inserting
+this into the DOM, the DOM parser will in turn see a regular element with an
+event handler.
+
+This kind of parse/re-parse mismatch (and its cousing, the parse context
+mismatch) has let to a number of attacks to bypass sanitizing libraryies,
+collectively known as mXSS.
+
+### Putting the Pieces Together: A New, Safer, and More Efficient API
+
+When contemplating these three concerns, the Sanitizer WICG decided to
+reformulate the Sanitizer API along these rules:
+
+- The API will return the DOM tree it operates on, thus discouraging re-parsing
+  in general.
+- All string-based operations require a context node that will be used for
+  HTML parsing.
+- The API will not "disappear" the context node. Either the operation will
+  operate on the context node directly, or it will pass along the context
+  node in the result.
+
+This leads to the current API:
+
+For `Sanitizer.sanitize`, no strings and therefore no HTML parsing are
+involved. It does not require a parsing context. It returns a modified DOM
+tree directly.
+
+For `Element.setHTML`, the context node is the `this` element that it is being
+called on. It returns no result at all, and instead the operation will
+commence on the `this` node.
+
+For `Sanitizer.sanitizeFor`, the developers will have to tell the Sanitizer
+what context they'd like to use the result in. The return value will contain
+both the result and the context node. Conveniently, a container for these two
+values already exists: It's the `Node` itself. Hence we return it.
+
+### What's Left Over?
+
+What if a developer wishes to have a string-to-string API after all? Maybe
+because an existing code base was built around the assumption of a
+string-to-string sanitization library, and it isn'd easy to refactor? In this
+case, the easiest way is to call `.sanitizeFor(value, context).innerHTML`, for
+a suitable context. However, we recommend against this practice, as this opens
+up the app to context- and re-parse-mismatches and therefore mXSS attacks.
+
```

Comment:
I wonder if we want this article to end on something more useful/actionable than this "what's left" section. Don't have a great suggestion though :sheep:

Reply:
I added a brief call for action, "Try it out!"